src/hooks/useWithdrawLiquidityHandler.tsx

```diff
@@ -1,360 +1,231 @@
+    try {
+      if (provider && accountInfo && accountInfo?.address) {
+        if (!gammaPoolContract && GAMMA_POOL_ADDRESS) {
+          setGammaPoolContract(
+            new ethers.Contract(
+              GAMMA_POOL_ADDRESS,
+              GammaPool.abi,
+              accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+            )
+          )
         }
       } else {
-        let _depPool = null
-
-        _depPool = new ethers.Contract(
-          address,
-          "",
-          accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
-        )
-        if (_depPool) {
-          setDepPool(_depPool)
-        }
+        notifyError('Please connect wallet')
       }
+    } catch (err) {
+      notifyError('An error occurred while initializing contract')
     }
-    fetchContract()
-  }, [provider])
-
-  useEffect(() => {
-    async function fetchData() {
-      if (process.env.NEXT_PUBLIC_ENVIRONMENT === 'local') {
-        if (!posManager) {
-          return
-        }
-        const liqBal = await posManager.balanceOf(accountInfo?.address)
-        setTotalLiquidityAmt(liqBal.toString())
-        setLiquidityAmt(parseFloat(ethers.utils.formatEther(liqBal)))
-      } else {
-        if (!depPool) {
-          return
-        }
-        const liqBal = await depPool.balanceOf(accountInfo?.address)
-        setTotalLiquidityAmt(liqBal.toString())
-        setLiquidityAmt(parseFloat(ethers.utils.formatEther(liqBal)))
-
-        const uniPair = await depPool.getUniPair()
-        if (!provider) {
-          return
-        }
-
-        const uniPairContract = new ethers.Contract(uniPair, IUniswapV2Pair.abi, provider)
-        const reserves = await uniPairContract.getReserves()
-        const _uniPrice = BigNumber.from(reserves.reserve1).mul(BigNumber.from(10).pow(18)).div(reserves.reserve0)
-        setUniPrice(_uniPrice.toString())
-        const liqBalNum = BigNumber.from(liqBal.toString())
+  }
 
-        if (liqBalNum.gt(constants.Zero) && _uniPrice.gt(constants.Zero)) {
-          setLiqInTokB(
-            parseFloat(
-              ethers.utils.formatEther(
-                sqrt(_uniPrice.mul(BigNumber.from(10).pow(18)))
-                  .mul(liqBalNum)
-                  .div(BigNumber.from(10).pow(18))
-                  .mul(2)
-              )
+  const initializePositionManagerContract = async () => {
+    try {
+      if (provider && accountInfo && accountInfo?.address) {
+        if (!positionManager && POSITION_MANAGER_ADDRESS) {
+          setPositionManager(
+            new ethers.Contract(
+              POSITION_MANAGER_ADDRESS,
+              PositionManager.abi,
+              accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
             )
           )
-        } else {
-          setLiqInTokB(0)
         }
+      } else {
+        notifyError('Please connect wallet')
```

Comment:
i think for a better user experience, we should set the confirm (disabled) button with a "Connect Wallet" message when this condition happens (more minimal), showing it as a notification is alarming to users

Reply:
Done